File: src/me/xerces/eventbus/event/Event.java
package me.xerces.eventbus.event;

/**
 * The Event class used as a base for other events
 * @author Xerces
 */
public abstract class Event {

    private boolean isCancelled;

    /**
     * setCancelled Will set if the event is cancelled or not
     * @param isCancelled Set if the event is cancelled, if it is the event manager will cease to fire it to lower priority handlers.
     */
    private void setCancelled(boolean isCancelled)
    {
        this.isCancelled = isCancelled;
    }

    /**
     * isCancelled check if the event has been cancelled, will stop the {@link me.xerces.eventbus.EventManager} from firing it to lower priority handlers.
     * @return isCancelled if the event is cancelled
     */
    private boolean isCancelled()
    {
        return isCancelled;
    }

}

File: src/main/java/dev/throwouterror/eventbus/event/Event.kt
package dev.throwouterror.eventbus.event

import org.json.JSONObject

/**
 * The Event class used as a base for other events
 */
open class Event {
    var type: String
        private set
    var parameters: JSONObject
        private set

    constructor(type: String, parameters: JSONObject? = JSONObject()) {
        this.type = type
        this.parameters = parameters!!
    }

    /**
     * isCancelled check if the events has been cancelled, will stop the [dev.throwouterror.eventbus.SimpleEventManager] from firing it to lower priority handlers.
     * @return isCancelled if the events is cancelled
     */
    var isCancelled = false
}